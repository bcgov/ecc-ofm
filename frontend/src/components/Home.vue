<!-- eslint-disable vue/multi-word-component-names -->
<template>
  <v-card
    title="OFM"
    subtitle="Temp Home Screen"
    text="..."
    style="height:100%;"
  ></v-card>
</template>
