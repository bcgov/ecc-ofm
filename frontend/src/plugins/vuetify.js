import '@fortawesome/fontawesome-free/css/all.css';
import '@mdi/font/css/materialdesignicons.css';
import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import { Ripple } from 'vuetify/lib/directives';

Vue.use(Vuetify, {
  directives: {
    Ripple
  }
});

export default new Vuetify({
  theme: {
    light: true,
    dark: false
  },
  icons: {
    iconfont: 'fa',
    values: {
      login: 'fas fa-user-clock',
      fast: 'fas fa-shipping-fast',
      sign_in: 'fas fa-sign-in-alt',
      info1: 'fas fa-info-circle',
      downArrow: 'fas fa-angle-down',
      upArrow: 'fas fa-angle-up',
      user: 'far fa-user',
      copy: 'fas fa-copy',
      search: 'fas fa-search',
      error: 'fas fa-exclamation-triangle',
      lock: 'fas fa-lock',
      info2: 'fas fa-info-circle fa-10x',
      question: 'fas fa-question-circle fa-10x',
      plus: 'fas fa-plus-circle',
      minus: 'fas fa-minus-circle',
    }
  }
});
