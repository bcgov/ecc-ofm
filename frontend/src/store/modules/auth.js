import ApiService from '@/common/apiService';
/* TODO: uncomment during integration with authservice...
import AuthService from '@/common/authService';
*/
import { Routes } from '@/utils/constants';
import {defineStore} from 'pinia';

export const authStore = defineStore('auth', {
  namespaced: true,
  state: () => ({
    acronyms: [],
    isAuthenticated: true  /* TODO: during integration with backend, replace true with... localStorage.getItem('jwtToken') !== null */,
    isAuthorizedUser: true /* TODO: during integration with backend, replace true with... localStorage.getItem('isAuthorizedUser') !== null */,
    userInfo: false,
    isValidChildCareProviderUser: localStorage.getItem('iisValidChildCareProviderUser') !== null,
    isValdProgramUser: localStorage.getItem('isValdProgramUser') !== null,
    isValidFinancialOpsUser: localStorage.getItem('isValidFinancialOpsUser') !== null,    
  }),
  getters: {
    acronymsGet: state => state.acronyms,
    isAuthenticatedGet: state => state.isAuthenticated,
    jwtTokenGet: () => localStorage.getItem('jwtToken'),
    userInfoGet: state => state.userInfo,
    //TODO: 3 temp roles ('CCP_ROLE', 'OPS_ROLE', 'PCM_ROLE') were created in auth.js (loosely
    //based on OFM requirements) for the purpose of achieving a 1st draft of the frontend that
    //will render a home screen and menu with minimal errors given no authorization/backend integration.
    //Thus the following related code is only temporarly and expected to be replaced...
    CCP_ROLE: state => state.isValidChildCareProviderUser,
    OPS_ROLE: state => state.isValdProgramUser,
    PCM_ROLE: state => state.isValidFinancialOpsUser
  },
  actions: {
    //sets Json web token and determines whether user is authenticated
    async setJwtToken(token = null) {
      if (token) {
        this.isAuthenticated = true;
        localStorage.setItem('jwtToken', token);
      } else {
        this.isAuthenticated = false;
        localStorage.removeItem('jwtToken');
      }
    },
    async setAuthorizedUser(isAdminUser){
      if (isAdminUser) {
        this.isAuthorizedUser = true;
        localStorage.setItem('isAuthorizedUser', 'true');
      } else {
        this.isAuthorizedUser = false;
        localStorage.removeItem(('isAuthorizedUser'));
      }
    },
    async setAuthorizedWebsocketUser(isAuthorizedWebsocketUser){
      if (isAuthorizedWebsocketUser) {
        this.isAuthorizedWebsocketUser = true;
        localStorage.setItem('isAuthorizedWebsocketUser', 'true');
      } else {
        this.isAuthorizedWebsocketUser = false;
        localStorage.removeItem(('isAuthorizedWebsocketUser'));
      }
    },
    async setUserInfo(userInf) {
      if (userInf) {
        this.userInfo = userInf;
      } else {
        this.userInfo = null;
      }
    },
    // TODO: method below is part of 1st draft of app framework and likely to be removed...
    async setChildCareProviderUser(isValidChildCareProviderUser) {
      if (isValidChildCareProviderUser) {
        this.isValidChildCareProviderUser = true;
        localStorage.setItem('isValidChildCareProviderUser', 'true');
      } else {
        this.isValidChildCareProviderUser = false;
        localStorage.removeItem(('isValidChildCareProviderUser'));
      }
    },
    // TODO: method below is part of 1st draft of app framework and likely to be removed...
    async setProgramUser(isValdProgramUser) {
      if (isValdProgramUser) {
        this.isValdProgramUser = true;
        localStorage.setItem('isValdProgramUser', 'true');
      } else {
        this.isValdProgramUser = false;
        localStorage.removeItem(('isValdProgramUser'));
      }
    },
      // TODO: method below is part of 1st draft of app framework and likely to be removed...
    async setFinancialOpsUser(isValidFinancialOpsUser) {
      if (isValidFinancialOpsUser) {
        this.isValidFinancialOpsUser = true;
        localStorage.setItem('isValidFinancialOpsUser', 'true');
      } else {
        this.isValidFinancialOpsUser = false;
        localStorage.removeItem(('isValidFinancialOpsUser'));
      }
    },
    //sets the token required for refreshing expired json web tokens
    async logoutState() {
      localStorage.removeItem('jwtToken');
      this.userInfo = false;
      this.isAuthenticated = false;
    },
    async getUserInfo() {
      const token = localStorage.getItem('jwtToken');

      if(!token){
        await this.getJwtToken();
      }

      if(localStorage.getItem('jwtToken')) {
        const response = await ApiService.apiAxios.get(Routes.USER);
        await this.setUserInfo(response.data);
      }
    },
    //retrieves the json web token from local storage. If not in local storage, retrieves it from API
    async getJwtToken() {
      try {
        const token = localStorage.getItem('jwtToken');
        if (this.isAuthenticated && !!token) {
          /* TODO: uncomment during integration with authservice/backend...
          const response = await AuthService.refreshAuthToken(token);
          */
         //TODO: uncomment 'response' below during integration with backend...
          await this.setAuthorizations(/*response*/);
        } else {
          /* TODO: uncomment during integration with backend...
          const response = await AuthService.getAuthToken();
          */
         //TODO: uncomment 'response' below during integration with backend...
          await this.setAuthorizations(/*response*/);
        }
      } catch (e) {
        // Remove tokens from localStorage and update state
        await this.setJwtToken(null);
        throw e;
      }
    },
    //TODO: uncomment 'response' below during integration with authorization/backend...
    async setAuthorizations(/*response*/) {
    /* TODO: uncomment during integration with backend...
      if (response.jwtFrontend) {
        await this.setJwtToken(response.jwtFrontend);
      }

      ApiService.setAuthHeader(response.jwtFrontend);
    */
      await this.setAuthorizedUser(true);
      await this.setChildCareProviderUser(true);
      await this.setProgramUser(true);
      await this.setFinancialOpsUser(true);
    }
  }
});
