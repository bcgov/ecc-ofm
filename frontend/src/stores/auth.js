import moment from 'moment'
import { defineStore } from 'pinia'

import ApiService from '@/common/apiService'
import AuthService from '@/common/authService'
import { useAppStore } from '@/stores/app'
import { APPLICATION_INTAKE_TYPES, OFM_PROGRAM_CODES, ROLES } from '@/utils/constants'

export const useAuthStore = defineStore('auth', {
  namespaced: true,
  state: () => ({
    isAuthenticated: localStorage.getItem('jwtToken') !== null,
    isMinistryUser: false,
    isImpersonating: false,
    isUserInfoLoaded: false,
    userInfo: null,
    impersonateId: null,
    currentFacility: {},
    permissions: [],
  }),
  getters: {
    isActingProvider: (state) => !state.isMinistryUser || state.isImpersonating,
    hasFacilities: (state) => state.userInfo?.facilities?.length > 0,
    hasPermission: (state) => {
      return (permission) => {
        return state.permissions?.some((p) => p === permission)
      }
    },
  },
  actions: {
    //sets Json web token and determines whether user is authenticated
    async setJwtToken(token = null) {
      if (token) {
        this.isAuthenticated = true
        localStorage.setItem('jwtToken', token)
      } else {
        this.isAuthenticated = false
        localStorage.removeItem('jwtToken')
      }
    },
    //sets the token required for refreshing expired json web tokens
    async logout() {
      localStorage.removeItem('jwtToken')
      this.userInfo = null
      this.isAuthenticated = false
    },
    async getUserInfo() {
      const token = localStorage.getItem('jwtToken')
      if (!token) {
        await this.getJwtToken()
      }
      if (localStorage.getItem('jwtToken')) {
        if (!this.isUserInfoLoaded) {
          const appStore = useAppStore()
          let userInfoRes = undefined
          if (this.impersonateId && this.isMinistryUser) {
            userInfoRes = await ApiService.getUserImpersonateInfo(this.impersonateId)
            this.isImpersonating = true
          } else {
            userInfoRes = await ApiService.getUserInfo()
            this.isMinistryUser = userInfoRes.data.isMinistryUser
            delete userInfoRes.data.isMinistryUser
          }
          this.userInfo = userInfoRes.data
          // Default the facility
          if (this.isActingProvider && this.hasFacilities) {
            this.currentFacility = this.userInfo.facilities[0]
          }
          
          // Lookup the permissions
          let role
          if (this.isImpersonating) {
            // When impersonating always use 'Impersonate', not the impersonated user's role
            role = appStore.roles.find((role) => role.roleName === ROLES.IMPERSONATE)
          } else {
            role = appStore.roles.find((role) => role.roleId === this.userInfo.role?.ofm_portal_roleid)
          }
          this.permissions = role?.permissions.map((p) => p.permissionName)

          /*
            A facility can apply for a core application if it satisfies these conditions:
            1. There is an Open Intake or the facility is in the allowed facility list of a Limited Intake
            2. If the facility is in the CCOF program, their program start date must >= 1 year from the current date.
          */
          this.userInfo?.facilities?.forEach((facility) => {
            facility.intakeWindowCheckForAddApplication = appStore.applicationIntakes?.some((intake) => {
              const isWithinApplicationIntakeWindow = moment().isSameOrAfter(moment(intake.startDate)) && moment().isSameOrBefore(moment(intake.endDate))
              const isOpenIntake = intake.type === APPLICATION_INTAKE_TYPES.OPEN_INTAKE
              const limitedIntakeFacilities = intake?.facilities?.map((facility) => facility.facilityId)
              return isWithinApplicationIntakeWindow && (isOpenIntake || limitedIntakeFacilities?.includes(facility.facilityId))
            })
            facility.ccofEnrolmentCheckForAddApplication = facility?.programCode !== OFM_PROGRAM_CODES.CCOF || (facility?.programCode === OFM_PROGRAM_CODES.CCOF && facility?.ccofOneYearEnrolment)
          })

          this.isUserInfoLoaded = true
        }
      }
    },
    //retrieves the json web token from local storage. If not in local storage, retrieves it from API
    async getJwtToken() {
      try {
        const token = localStorage.getItem('jwtToken')
        if (this.isAuthenticated && !!token) {
          const response = await AuthService.refreshAuthToken(token)
          await this.setAuthorizations(response)
        } else {
          const response = await AuthService.getAuthToken()
          await this.setAuthorizations(response)
        }
      } catch (e) {
        // Remove tokens from localStorage and update state
        await this.setJwtToken(null)
        throw e
      }
    },
    async setAuthorizations(response) {
      if (response.jwtFrontend) {
        await this.setJwtToken(response.jwtFrontend)
      }
      ApiService.setAuthHeader(response.jwtFrontend)
    },
  },
})
