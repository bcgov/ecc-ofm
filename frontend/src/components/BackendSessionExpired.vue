<template>
  <div style="display: none">
    <a id="logout_href" :href="link" />
  </div>
</template>

<script>
import { Routes } from '@/utils/constants'
import { useAuthStore } from '@/stores/auth'
import { mapActions } from 'pinia'
export default {
  name: 'BackendSessionExpired',
  data() {
    return {
      link: Routes.SESSION_EXPIRED
    }
  },
  mounted() {
    window.location = document.getElementById('logout_href').href
  },
  methods: {
    ...mapActions(useAuthStore, ['setJwtToken']),
    logout() {
      this.setJwtToken()
    }
  }
}
</script>
