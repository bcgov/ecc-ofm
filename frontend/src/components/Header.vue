<!-- eslint-disable vue/multi-word-component-names -->
<template>
  <v-app-bar absolute color="rgb(0, 51, 102)" class="sysBar pl-10 pr-10" style="z-index: 1002">
    <!-- Navbar content -->
    <a tabindex="-1" href="/">
      <img tabindex="-1" src="@/assets/images/bc-gov-logo.svg" class="logo" alt="B.C. Government Logo" />
    </a>
    <a tabindex="-1" href="/">
      <img class="verticalLine" />
    </a>
    <v-toolbar-title style="margin-left: 0px;">
      <h3>{{ appTitle }}</h3>
    </v-toolbar-title>
    <v-spacer></v-spacer>
    <div v-if="isAuthenticated && user">
      <v-menu name="user_options" offset-y>
        <template #activator="{ props }">
          <v-chip v-bind="props" tabindex="0" pill color="#003366" dark>
            <v-avatar left color="info">
              {{ user.userName[0] }}
            </v-avatar>
            <span class="display-name pl-1">{{ user.userName }}</span>
          </v-chip>
        </template>
        <v-list dark style="background-color: #003366; color: white">
          <v-list-item :href="routes.LOGOUT" id="logout_button" style="min-height: 4vh" title="Logout" />
        </v-list>
      </v-menu>
    </div>
  </v-app-bar>
</template>

<script>
import { mapActions, mapState } from 'pinia'
import { Routes } from '@/utils/constants'
import { useAuthStore } from '@/stores/auth'

export default {
  data() {
    return {
      appTitle: import.meta.env.VITE_APP_TITLE,
      routes: Routes,
      user: null
    }
  },
  created() {
      useAuthStore().getUserInfo().then(() => {
        this.user = this.userInfo
      })
  },
  computed: {
    ...mapState(useAuthStore, ['userInfo', 'isAuthenticated'])
  },
  methods: {
    ...mapActions(useAuthStore, ['getUserInfo']),
    useAuthStore
  }
}
</script>

<style>
.gov-header .v-icon {
  padding-left: 10px;
}

a {
  text-decoration: none;
}

.logo {
  padding-right: 15px;
  padding-top: 4px;
  width: 205px;
  height: 77px;
}

.verticalLine {
  border-left: 1px solid #DFB433;
  height: 50px;
  margin-left: 12px;
  padding-left: 24px;
  margin-top: 4px;
}

.gov-header .title {
  color: #fff;
  text-decoration: none;
}

.sysBar {
  border-bottom: 2px solid rgb(252, 186, 25) !important;
  z-index: 8;
}

.gov-header .v-btn,
.v-btn--active.title:before,
.v-btn.title:focus:before,
.v-btn.title:hover:before {
  color: #fff;
  background: none;
}

.secondary_color {
  background-color: var(--v-secondary-base);
}

.v-input__slot {
  padding-top: 10px;
}

.display-name {
  color: white;
}

.top-down {
  padding-top: 20px;
  height: 80%;
}

.mainTitle {
  font-size: 1.0rem;
}
</style>
