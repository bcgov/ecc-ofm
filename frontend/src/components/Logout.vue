<template>
  <v-container fluid class="full-height">
    <!-- login article -->
    <article name="logout-banner">
      <v-row align="center" justify="center">
        <v-col xs="10" sm="10" md="8" lg="4" xl="3">
          <v-card class="session-expired-card">
            <v-card-title class="gov-header">
              <h4 id="logout_text">Logged Out</h4>
            </v-card-title>
            <v-card-text id="logout_descriptor">
              <v-row style="margin: 0.3rem"> You have Logged out. </v-row>
              <a
                id="login-button"
                :href="routes.LOGIN"
                class="ma-1"
                dark
                color="#003366"
                @click="clearStorage"
                >Log In</a
              ><span>again if you wish to continue.</span>
            </v-card-text>
          </v-card>
        </v-col>
      </v-row>
    </article>
  </v-container>
</template>

<script>
import { mapActions } from 'pinia'
import { Routes } from '../utils/constants'
import { useAuthStore } from '@/stores/auth'

export default {
  // eslint-disable-next-line vue/multi-word-component-names
  name: 'Logout',

  data() {
    return {
      routes: Routes
    }
  },
  created() {
    this.setJwtToken()
  },
  methods: {
    ...mapActions(useAuthStore, ['setJwtToken']),
    clearStorage() {
      this.setJwtToken()
    }
  }
}
</script>

<style scoped>
.full-height {
  height: 100%;
}
.session-expired-card {
  margin-top: 15rem;
  width: 100%;
  background: #d9e7d8;
}
</style>
