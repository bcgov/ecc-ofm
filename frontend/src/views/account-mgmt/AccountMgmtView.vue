<template>
  <v-container fluid>
    <h1>Account Management</h1>
    <v-card min-width="350px" max-width="50%">
      <v-list>
        <v-list-item>
          <template v-slot:prepend>
            <v-icon>mdi-office-building</v-icon>
          </template>
          <router-link :to="{ name: 'manage-organization' }">Manage Organization/Facilities</router-link>
        </v-list-item>
        <v-list-item>
          <template v-slot:prepend>
            <v-icon>mdi-account-group</v-icon>
          </template>
          <router-link :to="{ name: 'manage-users' }">Manage Users</router-link>
        </v-list-item>
      </v-list>
    </v-card>
    <AppBackButton id="back-home-button" size="large" width="220px" :to="{ name: 'home' }">Home</AppBackButton>
  </v-container>
</template>

<script>
import AppBackButton from '@/components/ui/AppBackButton.vue'
import rolesMixin from '@/mixins/rolesMixin.js'
import { mapActions } from 'pinia'
import { useAuthStore } from '@/stores/auth'

export default {
  components: { AppBackButton },
  mixins: [rolesMixin],
  methods: {
    ...mapActions(useAuthStore, ['hasRole']),
  },
}
</script>
