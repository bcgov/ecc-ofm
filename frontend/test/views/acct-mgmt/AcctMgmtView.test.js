import { describe, expect, test } from 'vitest'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

import { useAuthStore } from '@/stores/auth.js'
import AccountMgmtView from '@/views/account-mgmt/AccountMgmtView.vue'
import { createTestingPinia } from '@pinia/testing'
import { mount } from '@vue/test-utils'

const vuetify = createVuetify({ components, directives })

describe('Manage Users screen', () => {
  const global = { global: { plugins: [createTestingPinia(), vuetify] } }
  const authStore = useAuthStore()

  test('Test Account Manager Links', async () => {
    authStore.userInfo = { userName: 'accountManager', role: 2 }
    const wrapper = mount(AccountMgmtView, global)
    expect(wrapper.text()).toContain('Manage Organization/Facilities')
    expect(wrapper.text()).toContain('Manage Users')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Test Admin Links', async () => {
    authStore.userInfo = { userName: 'admin', role: 1 }
    const wrapper = mount(AccountMgmtView, global)
    expect(wrapper.text()).toContain('Manage Organization/Facilities')
    expect(wrapper.text()).not.toContain('Manage Users')

    expect(wrapper.html()).toMatchSnapshot()
  })
})
