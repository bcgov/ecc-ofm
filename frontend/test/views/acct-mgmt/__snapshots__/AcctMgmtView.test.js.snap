// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Manage Users screen > Test Account Manager 1`] = `
"<div class=\\"v-container v-container--fluid v-locale--is-ltr\\">
  <h1>Account Management</h1>
  <div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\" style=\\"max-width: 50%; min-width: 350px;\\">
    <!---->
    <div class=\\"v-card__loader\\">
      <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
        <!---->
        <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
        <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <div class=\\"v-progress-linear__indeterminate\\">
            <div class=\\"v-progress-linear__indeterminate long\\"></div>
            <div class=\\"v-progress-linear__indeterminate short\\"></div>
          </div>
        </transition-stub>
        <!---->
      </div>
    </div>
    <!---->
    <!---->
    <div class=\\"v-list v-theme--light v-list--density-default v-list--one-line\\" tabindex=\\"0\\" role=\\"listbox\\">
      <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
        <!----><span class=\\"v-list-item__underlay\\"></span>
        <div class=\\"v-list-item__prepend\\"><i class=\\"mdi-office-building mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
          <div class=\\"v-list-item__spacer\\"></div>
        </div>
        <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
          <!---->
          <!---->
          <router-link to=\\"[object Object]\\">Manage Organization/Facilities</router-link>
        </div>
        <!---->
      </div>
      <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
        <!----><span class=\\"v-list-item__underlay\\"></span>
        <div class=\\"v-list-item__prepend\\"><i class=\\"mdi-account-group mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
          <div class=\\"v-list-item__spacer\\"></div>
        </div>
        <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
          <!---->
          <!---->
          <router-link to=\\"[object Object]\\">Manage Users</router-link>
        </div>
        <!---->
      </div>
    </div>
    <!---->
    <!----><span class=\\"v-card__underlay\\"></span>
  </div><a data-v-d22d1084=\\"\\" class=\\"v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--size-large v-btn--variant-elevated px-2 BC-Gov-SecondaryButton text-app-button mt-2 mt-2\\" style=\\"width: 220px;\\" id=\\"back-home-button\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><span data-v-d22d1084=\\"\\" class=\\"button-text overflow-hidden\\"><i class=\\"mdi-arrow-left mdi v-icon notranslate v-theme--light v-icon--size-default pb-1\\" aria-hidden=\\"true\\"></i> Back to Home</span>
    <!--v-if--></span>
    <!---->
    <!---->
  </a>
</div>"
`;

exports[`Manage Users screen > Test Account Manager Links 1`] = `
"<div class=\\"v-container v-container--fluid v-locale--is-ltr\\">
  <h1>Account Management</h1>
  <div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\" style=\\"max-width: 50%; min-width: 350px;\\">
    <!---->
    <div class=\\"v-card__loader\\">
      <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
        <!---->
        <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
        <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <div class=\\"v-progress-linear__indeterminate\\">
            <div class=\\"v-progress-linear__indeterminate long\\"></div>
            <div class=\\"v-progress-linear__indeterminate short\\"></div>
          </div>
        </transition-stub>
        <!---->
      </div>
    </div>
    <!---->
    <!---->
    <div class=\\"v-list v-theme--light v-list--density-default v-list--one-line\\" tabindex=\\"0\\" role=\\"listbox\\">
      <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
        <!----><span class=\\"v-list-item__underlay\\"></span>
        <div class=\\"v-list-item__prepend\\"><i class=\\"mdi-office-building mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
          <div class=\\"v-list-item__spacer\\"></div>
        </div>
        <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
          <!---->
          <!---->
          <router-link to=\\"[object Object]\\">Manage Organization/Facilities</router-link>
        </div>
        <!---->
      </div>
      <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
        <!----><span class=\\"v-list-item__underlay\\"></span>
        <div class=\\"v-list-item__prepend\\"><i class=\\"mdi-account-group mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
          <div class=\\"v-list-item__spacer\\"></div>
        </div>
        <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
          <!---->
          <!---->
          <router-link to=\\"[object Object]\\">Manage Users</router-link>
        </div>
        <!---->
      </div>
    </div>
    <!---->
    <!----><span class=\\"v-card__underlay\\"></span>
  </div><a data-v-d22d1084=\\"\\" class=\\"v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--size-large v-btn--variant-elevated px-2 BC-Gov-SecondaryButton text-app-button mt-2 mt-2\\" style=\\"width: 220px;\\" id=\\"back-home-button\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><span data-v-d22d1084=\\"\\" class=\\"button-text overflow-hidden\\"><i class=\\"mdi-arrow-left mdi v-icon notranslate v-theme--light v-icon--size-default pb-1\\" aria-hidden=\\"true\\"></i> Back to Home</span>
    <!--v-if--></span>
    <!---->
    <!---->
  </a>
</div>"
`;

exports[`Manage Users screen > Test Admin 1`] = `
"<div class=\\"v-container v-container--fluid v-locale--is-ltr\\">
  <h1>Account Management</h1>
  <div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\" style=\\"max-width: 50%; min-width: 350px;\\">
    <!---->
    <div class=\\"v-card__loader\\">
      <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
        <!---->
        <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
        <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <div class=\\"v-progress-linear__indeterminate\\">
            <div class=\\"v-progress-linear__indeterminate long\\"></div>
            <div class=\\"v-progress-linear__indeterminate short\\"></div>
          </div>
        </transition-stub>
        <!---->
      </div>
    </div>
    <!---->
    <!---->
    <div class=\\"v-list v-theme--light v-list--density-default v-list--one-line\\" tabindex=\\"0\\" role=\\"listbox\\">
      <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
        <!----><span class=\\"v-list-item__underlay\\"></span>
        <div class=\\"v-list-item__prepend\\"><i class=\\"mdi-office-building mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
          <div class=\\"v-list-item__spacer\\"></div>
        </div>
        <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
          <!---->
          <!---->
          <router-link to=\\"[object Object]\\">Manage Organization/Facilities</router-link>
        </div>
        <!---->
      </div>
      <!--v-if-->
    </div>
    <!---->
    <!----><span class=\\"v-card__underlay\\"></span>
  </div><a data-v-d22d1084=\\"\\" class=\\"v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--size-large v-btn--variant-elevated px-2 BC-Gov-SecondaryButton text-app-button mt-2 mt-2\\" style=\\"width: 220px;\\" id=\\"back-home-button\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><span data-v-d22d1084=\\"\\" class=\\"button-text overflow-hidden\\"><i class=\\"mdi-arrow-left mdi v-icon notranslate v-theme--light v-icon--size-default pb-1\\" aria-hidden=\\"true\\"></i> Back to Home</span>
    <!--v-if--></span>
    <!---->
    <!---->
  </a>
</div>"
`;

exports[`Manage Users screen > Test Admin Links 1`] = `
"<div class=\\"v-container v-container--fluid v-locale--is-ltr\\">
  <h1>Account Management</h1>
  <div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\" style=\\"max-width: 50%; min-width: 350px;\\">
    <!---->
    <div class=\\"v-card__loader\\">
      <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
        <!---->
        <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
        <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <div class=\\"v-progress-linear__indeterminate\\">
            <div class=\\"v-progress-linear__indeterminate long\\"></div>
            <div class=\\"v-progress-linear__indeterminate short\\"></div>
          </div>
        </transition-stub>
        <!---->
      </div>
    </div>
    <!---->
    <!---->
    <div class=\\"v-list v-theme--light v-list--density-default v-list--one-line\\" tabindex=\\"0\\" role=\\"listbox\\">
      <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
        <!----><span class=\\"v-list-item__underlay\\"></span>
        <div class=\\"v-list-item__prepend\\"><i class=\\"mdi-office-building mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
          <div class=\\"v-list-item__spacer\\"></div>
        </div>
        <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
          <!---->
          <!---->
          <router-link to=\\"[object Object]\\">Manage Organization/Facilities</router-link>
        </div>
        <!---->
      </div>
      <!--v-if-->
    </div>
    <!---->
    <!----><span class=\\"v-card__underlay\\"></span>
  </div><a data-v-d22d1084=\\"\\" class=\\"v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--size-large v-btn--variant-elevated px-2 BC-Gov-SecondaryButton text-app-button mt-2 mt-2\\" style=\\"width: 220px;\\" id=\\"back-home-button\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><span data-v-d22d1084=\\"\\" class=\\"button-text overflow-hidden\\"><i class=\\"mdi-arrow-left mdi v-icon notranslate v-theme--light v-icon--size-default pb-1\\" aria-hidden=\\"true\\"></i> Back to Home</span>
    <!--v-if--></span>
    <!---->
    <!---->
  </a>
</div>"
`;
